libs/nx-remotecache-s3/src/lib/s3-client.ts

```diff
@@ -5,20 +5,35 @@ import { getDefaultRoleAssumerWithWebIdentity } from '@aws-sdk/client-sts';
 
 import type { S3ClientConfig } from '@aws-sdk/client-s3/dist-types/S3Client';
 import type { DefaultProviderInit } from '@aws-sdk/credential-provider-node/dist-types/defaultProvider';
+import type { AwsCredentialIdentity, Provider } from '@aws-sdk/types';
 
 const ENV_ENDPOINT = 'NXCACHE_S3_ENDPOINT';
 const ENV_PROFILE = 'NXCACHE_S3_PROFILE';
 const ENV_FORCE_PATH_STYLE = 'NXCACHE_S3_FORCE_PATH_STYLE';
 const ENV_REGION = 'NXCACHE_S3_REGION';
+const ENV_ACCESS_KEY_ID = 'NXCACHE_S3_ACCESS_KEY_ID';
+const ENV_SECRET_ACCESS_KEY = 'NXCACHE_S3_SECRET_ACCESS_KEY';
 
 export const buildS3Client = (
   options: Pick<S3ClientConfig, 'endpoint' | 'region' | 'forcePathStyle'> &
     Pick<DefaultProviderInit, 'profile'>
 ) => {
-  const provider = defaultProvider({
-    profile: getEnv(ENV_PROFILE) ?? options.profile,
-    roleAssumerWithWebIdentity: getDefaultRoleAssumerWithWebIdentity(),
-  });
+  const awsAccessKeyIdOverride = getEnv(ENV_ACCESS_KEY_ID);
```

Comment:
Personally I would extract this logic related to building the credentials provider into a separate function. This way we can keep it simple in here and just have a const variable `const provider = buildCredentials();`